server/plugin/api.go

```diff
@@ -1187,6 +1193,66 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
 	returnStatusOK(w)
 }
 
+func (p *Plugin) handlePipelineCommentModal(w http.ResponseWriter, r *http.Request) {
+	mattermostUserID := r.Header.Get(constants.HeaderMattermostUserID)
+	decoder := json.NewDecoder(r.Body)
+	postActionIntegrationRequest := &model.PostActionIntegrationRequest{}
+	if err := decoder.Decode(&postActionIntegrationRequest); err != nil {
+		// TODO: prevent posting any error messages except oAuth in DM for now and use dialog for all such cases
+		p.handlePipelineApprovalRequestUpdateError("Error decoding PostActionIntegrationRequest param: ", mattermostUserID, err)
+		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: err.Error()})
+		return
+	}
+
+	var project, approval, url string
+	if postActionIntegrationRequest.Context[constants.PipelineRequestContextRequestName].(string) == constants.PipelineRequestNameRelease {
+		url = fmt.Sprintf("%s%s", p.GetPluginURL(), constants.PathPipelineReleaseRequest)
+		approval = fmt.Sprintf("%f", postActionIntegrationRequest.Context[constants.PipelineRequestContextApprovalID].(float64))
+		project = postActionIntegrationRequest.Context[constants.PipelineRequestContextProjectName].(string)
+	} else {
+		url = fmt.Sprintf("%s%s", p.GetPluginURL(), constants.PathPipelineRunRequest)
+		approval = postActionIntegrationRequest.Context[constants.PipelineRequestContextApprovalID].(string)
+		project = postActionIntegrationRequest.Context[constants.PipelineRequestContextProjectID].(string)
+	}
+
+	organization := postActionIntegrationRequest.Context[constants.PipelineRequestContextOrganization].(string)
+	requestType := postActionIntegrationRequest.Context[constants.PipelineRequestContextRequestType].(string)
+
+	elements := []model.DialogElement{
+		{
+			DisplayName: "Comment:",
+			Name:        constants.DialogFieldNameComment,
+			Type:        "text",
+			Optional:    true,
```

Comment:
whats the default value for optional and check if it is true remove this line

Reply:
default value is false